lib/connectors_async/job.rb

```diff
@@ -86,5 +92,18 @@ def is_finished?
 
       [ConnectorsShared::JobStatus::FINISHED, ConnectorsShared::JobStatus::FAILED].include?(status)
     end
+
+    def safe_to_clean_up?
+      return true if is_finished? && @data[:documents].empty?
```

Comment:
That's true, however Enterprise Search also consumes the documents in the queue even from the FAILED job. However, I think for this logic here it might make sense to ignore the state of the queue.

Reply:
Hmmmm no, actually the job can be FINISHED, but the documents still not collected (state machine logic is a bit awkward here).

We need to make sure that the queue is empty, because Enterprise Search will stop only if both queue is empty and status is finished.

This logic can be improved by making state machine more robust, and we'll approach it in future.